File: SegundaParte/Semana7/Java/HerancaSimples/Serie.java
public abstract class Serie {
    protected int position;
    protected int value;

    public Serie(int position) {
        this.position = position;
        this.value = 0;
    }

    public void newPosition(int newPosition) {
        this.position = newPosition;
    }

    abstract public int compute(); // <<-- Método abstrato

    public String get() {
        return "Position: " + position + ", Value: " + value;
    }
}

File: SegundaParte/Semana7/C++/Serie.cpp
#include "Serie.h"

Serie::Serie(int position) : position(position), value(0) {}

void Serie::newPosition(int newPosition) {
    position = newPosition;
}

std::string Serie::get() {
    return "Position: " + std::to_string(position) + ", Value: " + std::to_string(value);
}

